### PR TITLE
feat: add kommander operator scaffold

### PR DESCRIPTION
**Pull request by Azhovan**
**What problem does this PR solve?**:

This PR scaffolds the necessary kubernetes resouces that are required to successfully deploy **Kommander Operator**. 
The Kommander Operator is an operator that manage Kommander lifecycle custom resource 
for more information[ visit this KEP](https://github.com/mesosphere/dkp-platform/pull/376/files?short_path=2d954ab#diff-2d954aba9b1a228591c080d531d8db6adad1271d4053abf0b57c54d0288895d9)

**Which issue(s) does this PR fix?**:
https://d2iq.atlassian.net/browse/D2IQ-95249

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

**Review by palexster — APPROVED** [review]

File: charts/kommander-operator/templates/post_install_kommandercore_hook.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+    kubernetes.io/enforce-mountable-secrets: "true"
+secrets:
+  - name: {{ .Chart.Name }}-installation
+
+---
+apiVersion: v1
```
Comment (mikolajb) [inline charts/kommander-operator/templates/post_install_kommandercore_hook.yaml:16]: Do we really need the secret?

File: charts/kommander-operator/templates/rback_v1_clusterrole_operator_kommander-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      - apps
+      - clusterapps
+      - helmreleasestatuses
+      - kommanderclusters
+      - virtualgroups
+      - projects
+      - workspaces
+      - secrets
+      - configmaps
+      - federatedconfigmaps
```
Comment (mikolajb) [inline charts/kommander-operator/templates/rback_v1_clusterrole_operator_kommander-operator.yaml:106]: We can probably remove a few items from here:
```suggestion
      - appdeployments
      - apps
      - clusterapps
      - secrets
      - configmaps
      - federatedconfigmaps
```

File: charts/kommander-operator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+replicaCount: 1
+image:
+  repository: nginx
+  tag: 0.1.0
+  pullPolicy: IfNotPresent
+
+kubetools:
+  image:
+    repository: mesosphere/kommander2-kubetools
+    tag: v2.4.0
```
Comment (mikolajb) [inline charts/kommander-operator/values.yaml:13]: Could you add a note (and maybe a ticket in jira) so we don't forget to bind it with the kommander version?

File: charts/kommander-operator/templates/post_install_kommandercore_hook.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        echo "Waiting for the secret to be populated with credentials"
+        sleep 1
+    done
+
+    cat <<EOF | kubectl apply -f -
+    apiVersion: dkp.d2iq.io/v1alpha1
+    kind: KommanderCore
+    metadata:
+      name: kommander-core
+      namespace: kommander
```
Comment (cwyl02) [inline charts/kommander-operator/templates/post_install_kommandercore_hook.yaml:73]: Should we remove the NS here? I thought this is a cluster resource. @palexster

Comment (palexster) [reply charts/kommander-operator/templates/post_install_kommandercore_hook.yaml:73]: Yes, I fixed the KEP PR accordingly. Thanks!

File: charts/kommander-operator/templates/post_install_kommandercore_hook.yaml (hunk 1 above)
Comment (Azhovan) [reply charts/kommander-operator/templates/post_install_kommandercore_hook.yaml]: removed.

**Comment by Azhovan** [comment]
@palexster @mikolajb 
do we perhaps also need a restrictive network policy?

**Comment by palexster** [comment]
> @palexster @mikolajb do we perhaps also need a restrictive network policy here? thinking

The controller does not expose any port anything but the metrics one, which is not exposed by default on 0.0.0.0.

File: charts/kommander-operator/templates/rback_v1_clusterrole_operator_kommander-operator.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Chart.Name }}-view
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+  - apiGroups:
+        - apps.kommander.d2iq.io
```
Comment (mikolajb) [inline charts/kommander-operator/templates/rback_v1_clusterrole_operator_kommander-operator.yaml:22]: different indentation than the rest of the file

File: charts/kommander-operator/values.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+kommanderoperator:
+  image:
+    repository: nginx
+    tag: 1.22
+    pullPolicy: IfNotPresent
+
+kubetools:
+  image:
+    repository: mesosphere/kommander2-kubetools
+    tag: v2.5.0-dev # we should bump this version when kommander released or new kubernetes version is supported
```
Comment (mikolajb) [inline charts/kommander-operator/values.yaml:14]:
```suggestion
    tag: v2.5.0-dev # TODO(jira ticket 🙏🏻 ): align version with the kommander release (either by using release automation or releasing the chart with kommander)
```

**Review by mikolajb — APPROVED** [review]

**Pull request merged**